<a href="https://colab.research.google.com/github/thethinkmachine/DielectronAnalysis/blob/main/MonteCarloProbabilityEstimation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***Estimating the Probability of Rolling a Sum of 7 Using Monte Carlo Probability Estimation***

## **The Experiment: In Theory**
A standard six-sided die has faces numbered from 1 to 6. When rolling two dice, each die is independent of the other, so the total number of possible outcomes is the product of the number of faces on each die:

$$ 6 \times 6 = 36 $$

Each pair of outcomes $die1$, $die2$ is equally likely, and there are 36 such pairs.

Next, we identify all pairs of dice rolls ($die1$, $die2$) that sum to 7. These pairs are:

1. (1, 6)
2. (2, 5)
3. (3, 4)
4. (4, 3)
5. (5, 2)
6. (6, 1)

Each of these pairs results in a sum of 7. There are 6 such pairs.

### **Calculating the Probability**

The probability of an event is given by the ratio of the number of successful outcomes to the total number of possible outcomes. In this case:

- **Successful outcomes (sum equals 7)**: 6
- **Total possible outcomes (all dice combinations)**: 36

Thus, the probability \( P \) of rolling a sum of 7 with two dice is:

$$ P(\text{sum of 7}) = \frac{\text{Number of successful outcomes}}{\text{Total number of possible outcomes}} = \frac{6}{36} = \frac{1}{6} \approx 0.1667 $$

This means that there is approximately a 16.67% chance of rolling a sum of 7 with two six-sided dice.

# **The Experiment: In Practice**

## 1. Setting up Colab environment

*   We initialize the simulation environment by importing Numpy, which will be used to perform numerical operations and visualize the results.



In [1]:
# Import necessary libraries
import numpy as np

# 2. Defining the simulation
* We create a function that simulates rolling two six-sided dice and returns the sum of the numbers. This function will be used repeatedly to generate a large number of dice rolls.

In [2]:
# Function to simulate the event
def simulate_dice_roll():
    die1 = np.random.randint(1, 7)
    die2 = np.random.randint(1, 7)
    return die1 + die2 == 7


# 3. Runing Monte Carlo Simulations & Computing Estimated Probability
* We simulate rolling the two dice a large number of times (e.g., 100,000 times) and record the sum of the numbers for each roll. This step generates the data needed for estimating the probability.
* We then count the number of times the sum of the dice is 7. This count represents the number of successful outcomes where the desired event (sum of 7) occurs.
* Lastly, we calculate the estimated probability by dividing the number of successful outcomes by the total number of simulations. This ratio gives the empirical probability of rolling a sum of 7.

In [7]:
# Number of simulations
num_simulations = 1000000

# Run simulations and count successes
success_count = 0
for i in range(num_simulations):
    if simulate_dice_roll():
        success_count += 1

# Estimate the probability
estimated_probability = success_count / num_simulations
print(f"Estimated Probability: {estimated_probability}")


Estimated Probability: 0.16629


# **Result**

The estimated probability from the simulation was 0.166.., which is remarkably close to the theoretical probability of 0.1667.

# **Explanation for the observed result**

The theoretical probability of rolling a sum of 7 with two six-sided dice can be calculated using combinations. There are 6 successful outcomes $1+6, 2+5, 3+4, 4+3, 5+2, 6+1$ out of a total of 36 possible outcomes when rolling two dice. Thus, the theoretical probability is:

$$ P(\text{sum of 7}) = \frac{6}{36} = \frac{1}{6} \approx 0.1667 $$

Monte Carlo simulations rely on random sampling to estimate probabilities. In our experiment, we conducted 100,000 iterations, where each iteration simulated rolling two dice and checking if their sum was 7. The frequency of obtaining a sum of 7 in these simulations was divided by the total number of iterations to yield the estimated probability.

The reason our estimated probability (0.166..) is very close to the theoretical probability (0.1667) can be attributed to the Law of Large Numbers. This statistical principle states that as the number of trials in a random experiment increases, the empirical probability (observed in the simulations) tends to converge to the theoretical probability.

While there is always some degree of random variation in Monte Carlo simulations, increasing the number of iterations reduces the impact of this variation. In our case, performing 100,000 iterations provided a sufficiently large sample size, ensuring that the random fluctuations averaged out and the estimated probability closely matched the theoretical value.

___